pkg/3scale/amp/component/system.go

```diff
@@ -165,15 +181,30 @@ func (system *System) getSystemSMTPEnvsFromSMTPSecret() []v1.EnvVar {
 
 func (system *System) SystemRedisEnvVars() []v1.EnvVar {
 	result := []v1.EnvVar{}
-
 	result = append(result,
 		helper.EnvVarFromSecret("REDIS_URL", SystemSecretSystemRedisSecretName, SystemSecretSystemRedisURLFieldName),
 		helper.EnvVarFromSecret("REDIS_SENTINEL_HOSTS", SystemSecretSystemRedisSecretName, SystemSecretSystemRedisSentinelHosts),
 		helper.EnvVarFromSecret("REDIS_SENTINEL_ROLE", SystemSecretSystemRedisSecretName, SystemSecretSystemRedisSentinelRole),
 	)
+	if system.Options.RedisTLSEnabled {
+		result = append(result, system.SystemRedisTLSEnvVars()...)
+	} else {
+		result = append(result, helper.EnvVarFromValue("REDIS_SSL", "0"))
+	}
 
 	return result
 }
+func (system *System) sidekiqInitContainerRedisEnvVars() []v1.EnvVar {
+	result := []v1.EnvVar{}
+	result = append(result, system.SystemRedisEnvVars()...)
+
+	if system.Options.RedisTLSEnabled {
+		result = append(result, system.BackendRedisTLSEnvVars()...)
```

Comment:
so we are setting backend redis envvars here, which to me indicates there's a clear requirement on all redis being TLS or none? if so, I would simplify it a bit

Reply:
Yes, System require TLS in backend-redis secret